modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/NuxeoPrincipal.java

```diff
@@ -163,6 +168,13 @@
      */
     boolean isAdministrator();
 
+    /**
+     * @since 11.4 Used in the case where we don't want to fetch the references, where we won't be able to return
+     *        complete informations on this principal (the missing attributes will be if the principal is an
+     *        Administrator and the groups he is in).
+     */
+    boolean isComplete();
```

Comment:
Adding this kind of method on public and important API like that is an architecture change that should be discussed first. Please open a discussion on Slack in `#team-platform` about this, because I'm really not sure it's the best way to move forward.

Reply:
Cc @nuxeo/platform